WcaOnRails/app/controllers/api/v0/user_roles_controller.rb

```diff
@@ -8,11 +8,23 @@ class Api::V0::UserRolesController < Api::V0::ApiController
     end
   end
 
-  # The order in which the roles should be sorted.
-  STATUS_SORTING_ORDER = ['leader', 'senior_member', 'member'].freeze
-
-  private def status_sort_rank(status)
-    STATUS_SORTING_ORDER.find_index(status) || STATUS_SORTING_ORDER.length
+  private def sort_rank(group_type, status)
```

Comment:
Actually doing two sorts will not work because status of two different group types cannot be compared - example board's member should always be higher than team_committee's senior member.

But I was able to find another solution which should be clean. Now, I don't have 'rank', instead sorts based on multiple parameters: lead, eligible_voter, groupTypeRank, status and finally groupName to sort alphabetically if everything else is same.